activejob/lib/active_job/log_subscriber.rb

```diff
@@ -189,15 +189,25 @@ def error(progname = nil, &block)
       end
 
       def log_enqueue_source
-        source = extract_enqueue_source_location(caller)
+        source = enqueue_source_location
 
         if source
           logger.info("↳ #{source}")
         end
       end
 
-      def extract_enqueue_source_location(locations)
-        backtrace_cleaner.clean(locations.lazy).first
+      if Thread.respond_to?(:each_caller_location)
+        def enqueue_source_location
+          Thread.each_caller_location do |location|
+            frame = backtrace_cleaner.clean_frame(location)
+            return frame if frame
+          end
+          nil
+        end
+      else
+        def enqueue_source_location
+          backtrace_cleaner.clean(caller(1).lazy).first
```

Comment:
```suggestion
          backtrace_cleaner.clean(caller(2).lazy).first
```

`caller(1)` is the same as `caller()`, since you are one method deeper, you want `caller(2)`.

But also given `log_enqueue` source doesn't do much more than that, it might make sense to just inline this.